opam-repo-ci-service.opam

```diff
@@ -39,12 +38,51 @@ depends: [
   "conf-libev"
   "dockerfile" {>= "6.3.0"}
   "dockerfile-opam" {>= "6.3.0"}
-  "ocaml-version" {>= "2.4.0"}
   "mula" {>= "0.1.2"}
   "alcotest" {>= "1.0.0" & with-test}
   "alcotest-lwt" {>= "1.0.1" & with-test}
   "odoc" {with-doc}
+
+  # pin depends
+  ## current
+  "current"
+  "current_docker"
+  "current_git"
+  "current_github"
+  "current_rpc"
+  "current_web"
+
+  ## ocluster
+  "ocluster-api"
+  "current_ocluster"
+
+  ## ocaml-dockerfile
+  "dockerfile"
+  "dockerfile-opam"
+
+  "ocaml-version"
+
+  "obuilder-spec"
+]
+pin-depends: [
+  ["current.0.6.6~dev"        "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+  ["current_docker.0.6.6~dev" "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+  ["current_git.0.6.6~dev"    "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+  ["current_github.0.6.6~dev" "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+  ["current_rpc.0.6.6~dev"    "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+  ["current_web.0.6.6~dev"    "git+https://github.com/ocurrent/ocurrent.git#e43b6c5c896a4cc5ca5c1d0cfbf10a6401fbcd47"]
+
+  ["ocluster-api.0.3.0~dev"      "git+https://github.com/ocurrent/ocluster.git#912ecfd38f003c6f537abca7e5313973b15ba615"]
+  ["current_ocluster.0.3.0~dev"  "git+https://github.com/ocurrent/ocluster.git#912ecfd38f003c6f537abca7e5313973b15ba615"]
+
+  ["dockerfile.8.2.1~dev"       "git+https://github.com/ocurrent/ocaml-dockerfile.git#85891cad42189a03960b48792293e8fb3a18552c"]
+  ["dockerfile-opam.8.2.1~dev"  "git+https://github.com/ocurrent/ocaml-dockerfile.git#85891cad42189a03960b48792293e8fb3a18552c"]
+
+  ["ocaml-version.3.6.7~dev"  "git+https://github.com/ocurrent/ocaml-version.git#b80aad5425c7dd6f1e6f8925f5719c72f8b8f7fa"]
+
+  ["obuilder-spec.0.6.0~dev" "git+https://github.com/ocurrent/obuilder.git#c0eb046f52ca0c5fd9b560d4bdc0cb11c852d4f7"]
```

Comment:
We wouldn't need commit hashes on all these, but ocam-ci has a constraint requiring a has. We can fix this in a followup.

See https://github.com/ocurrent/ocaml-ci/blob/a29bd00fcea21bcbcc1876d693b356e84f9c1236/lib/pin_depends.ml#L96-L102